## DistanceCalculator.cfg
# Dynamic reconfigure parameters for DistanceCalculator

PACKAGE = "lidar_inten_detect"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("intensity_threshold", int_t, 0, "Intensity threshold for lidar", 40, 0, 255)
gen.add("proximity_threshold", double_t, 0, "Proximity threshold for grouping", 0.3, 0.0, 1.0)
gen.add("min_distance", double_t, 0, "Minimum distance between points", 0.6, 0.0, 5.0)
gen.add("max_distance", double_t, 0, "Maximum distance between points", 0.65, 0.0, 5.0)
gen.add("offset_length", double_t, 0, "Offset length for calculations", 0.64, 0.0, 5.0)
gen.add("offset_standby", double_t, 0, "Offset standby distance", 1.0, 0.0, 5.0)
gen.add("x_min", double_t, 0, "x_min", -0.3, -10, 10.0)
gen.add("x_max", double_t, 0, "x_max", -1.25 , -10, 10.0)
gen.add("y_min", double_t, 0, "y_min", -0.5, -10, 10.0)
gen.add("y_max", double_t, 0, "y_max", 0.5, -10, 10.0)
gen.add("min_sample", int_t, 0, "min_sample", 3, 0, 10)

exit(gen.generate(PACKAGE, "cfg", "DistanceCalculator"))